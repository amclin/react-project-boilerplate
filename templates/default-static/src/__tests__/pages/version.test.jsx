import React from 'react'
import { render } from '@testing-library/react'
import VersionPage from '../../pages/version'

jest.mock('next/config', () => () => ({
  publicRuntimeConfig: {
    appEnv: 'TEST'
  }
}))

describe('Pages:Version', () => {
  it('renders the version page', () => {
    const { asFragment, getByTestId } = render(<VersionPage />)
    const versionObj = JSON.parse(getByTestId('version-json').textContent)

    expect(versionObj.appEnv).toBe('TEST')
    expect(asFragment()).toMatchSnapshot()
  })
})
