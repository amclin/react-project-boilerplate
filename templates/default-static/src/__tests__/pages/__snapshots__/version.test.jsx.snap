// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages:Version renders the version page 1`] = `
<DocumentFragment>
  <div
    data-testid="version-json"
    id="version-json"
  >
    {"appEnv":"TEST","paths":["/index.html","example/second-page.html","version.html"]}
  </div>
</DocumentFragment>
`;
